compiler/hash-parser/src/operator.rs

```diff
@@ -47,121 +56,217 @@ impl Operator {
             return (None, 0);
         }
 
-        match &(token.unwrap()).kind {
-            // Since the 'as' keyword is also a binary operator, we have to handle it here...
-            TokenKind::Atom(TokenAtom::Keyword(Keyword::As)) => (Some(Operator::As), 1),
-            TokenKind::Atom(TokenAtom::Eq) => match gen.peek_second() {
-                Some(token) if token.kind == TokenKind::Atom(TokenAtom::Eq) => {
-                    (Some(Operator::EqEq), 2)
-                }
-                _ => (None, 0),
-            },
-            TokenKind::Atom(TokenAtom::Lt) => match gen.peek_second() {
-                Some(token) if token.kind == TokenKind::Atom(TokenAtom::Eq) => {
-                    (Some(Operator::LtEq), 2)
-                }
-                Some(token) if token.kind == TokenKind::Atom(TokenAtom::Lt) => {
-                    (Some(Operator::Shl), 2)
-                }
-                _ => (Some(Operator::Lt), 1),
-            },
-            TokenKind::Atom(TokenAtom::Gt) => match gen.peek_second() {
-                Some(token) if token.kind == TokenKind::Atom(TokenAtom::Eq) => {
-                    (Some(Operator::GtEq), 2)
-                }
-                Some(token) if token.kind == TokenKind::Atom(TokenAtom::Gt) => {
-                    (Some(Operator::Shr), 2)
+        let (op, mut consumed): (_, u8) = match &(token.unwrap()).kind {
```

Comment:
See comment above about usize return type

Reply:
Given that how many times this function will be called, i think it's reasonable to keep the return type size as a u8.